project1/client/CMakeLists.txt

```diff
@@ -0,0 +1,3 @@
+add_subdirectory(input_parser)
+add_library(client input_parser/input_parser.h client.cpp client.h client_main.cpp)
```

Comment:
I don't think you want `client_main.cpp` to be part of this library, right? You probably want to remove that and add something like
```
add_executable(client_exe client_main.cpp)
target_link_libraries(client_main client)
```

Although, for consistency with the server, I'd call the library `client_lib` or something, and the executable `client`.